src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -948,12 +938,18 @@ impl LinkCollector<'_, '_> {
             return None;
         }
 
-        let cx = self.cx;
```

Comment:
This one is necessary.
```
error[E0502]: cannot borrow `*self` as mutable because it is also borrowed as immutable
    --> src/librustdoc/passes/collect_intra_doc_links.rs:1082:39
     |
942  |         let cx = &self.cx;
     |                  -------- immutable borrow occurs here
...
1082 |         let (mut res, mut fragment) = self.resolve_with_disambiguator_cached(
     |                                       ^^^^ mutable borrow occurs here
...
1106 |                             cx,
     |                             -- immutable borrow later used here
```